### PR TITLE
Use Spark version as demo-base docker image tag

### PR DESCRIPTION
**Pull request by lresende**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,19 +125,28 @@ docker-images:  ## Build docker images (includes kernel-based images)
 kernel-images: ## Build kernel-based docker images
 
 # Actual working targets...
-docker-images enterprise-gateway-demo demo-base nb2kg kernel-images enterprise-gateway kernel-py kernel-spark-py kernel-r kernel-spark-r kernel-scala kernel-tf-py kernel-tf-gpu-py kernel-image-puller:
+docker-images enterprise-gateway-demo nb2kg kernel-images enterprise-gateway kernel-py kernel-spark-py kernel-r kernel-spark-r kernel-scala kernel-tf-py kernel-tf-gpu-py kernel-image-puller:
 	make WHEEL_FILE=$(WHEEL_FILE) VERSION=$(VERSION) NO_CACHE=$(NO_CACHE) TAG=$(TAG) -C etc $@
```
Comment (kevin-bates) [inline Makefile:129]: I think we might want to "flow" SPARK_VERSION here so we can reference it in the eg-demo/Dockerfile.
```suggestion
	make WHEEL_FILE=$(WHEEL_FILE) VERSION=$(VERSION) NO_CACHE=$(NO_CACHE) TAG=$(TAG) SPARK_VERSION=$(SPARK_VERSION) -C etc $@
```

and change [this line of etc/Makefile](https://github.com/jupyter/enterprise_gateway/blob/master/etc/Makefile#L153) to specify SPARK_VERSION as a build-arg:
```
(cd ../build/docker/$1; docker build ${NO_CACHE} --build-arg HUB_ORG=${HUB_ORG} --build-arg TAG=${TAG} --build-arg SPARK_VERSION=${SPARK_VERSION} -t $(HUB_ORG)/$1:$(TAG) .)
```

File: etc/docker/enterprise-gateway-demo/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ARG HUB_ORG
-ARG TAG
```
Comment (kevin-bates) [inline etc/docker/enterprise-gateway-demo/Dockerfile:2]: We should add `ARG SPARK_VERSION` then, see next comment.

File: etc/docker/enterprise-gateway-demo/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,6 @@
...
 
-FROM $HUB_ORG/demo-base:$TAG
+FROM $HUB_ORG/demo-base:2.4.1
```
Comment (kevin-bates) [inline etc/docker/enterprise-gateway-demo/Dockerfile:3]: Per previous comments, we could just reference SPARK_VERSION here...
```suggestion
FROM $HUB_ORG/demo-base:$SPARK_VERSION
```

**Review by kevin-bates — COMMENTED** [review]
Good stuff.  Had a suggestion for how we could avoid having to update enterprise-gateway-demo/Dockerfile when the spark version changes.

**Comment by lresende** [comment]
Thank you @kevin-bates, indeed better to flow the spark version and have one less place to update when spark version changes. I have made those updates now.

**Review by kevin-bates — APPROVED** [review]
These changes look good - thank you.

**Pull request merged**